### PR TITLE
pangea-sdk: update Vault APIs to v2 (PAN-16797)

### PR DESCRIPTION
**Pull request by kenany**
- [x] Update paths from `v2beta` to `v2`.
- [x] Vault v2 is in LVE.
- [x] Revert CI to run on LVE.

### CODE REVIEW

File: packages/pangea-sdk/.sdk-ci.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@
     SERVICE_FILE_INTEL_ENV: LVE
     SERVICE_USER_INTEL_ENV: LVE
     SERVICE_REDACT_ENV: LVE
-    SERVICE_VAULT_ENV: LVE
+    SERVICE_VAULT_ENV: DEV
```
Comment (kenany) [inline packages/pangea-sdk/.sdk-ci.yml:27]: Undo this when Vault v2 is in LVE.

File: packages/pangea-sdk/pangea/asyncio/services/vault.py [hunk 2]
```diff
@@ -129,120 +142,118 @@ def __init__(
...
-    # Delete endpoint
-    async def delete(self, id: str) -> PangeaResponse[DeleteResult]:
+    async def delete(self, item_id: str, *, recursive: bool = False) -> PangeaResponse[DeleteResult]:
         """
         Delete
 
         Delete a secret or key
 
-        OperationId: vault_post_v1_delete
+        OperationId: vault_post_v2beta_delete
```
Comment (pangea-andrest) [inline packages/pangea-sdk/pangea/asyncio/services/vault.py:151]: operation should be updated to v2 as same as paths

File: packages/pangea-sdk/pangea/asyncio/services/vault.py [hunk 3]
```diff
@@ -321,719 +329,1422 @@ async def update(
...
+                name=name,
+                folder=folder,
+                metadata=metadata,
+                tags=tags,
+                disabled_at=disabled_at,
+                **kwargs,
+            ),
+        )
+
+    async def store_secret(
```
Comment (pangea-andrest) [inline packages/pangea-sdk/pangea/asyncio/services/vault.py:377]: Maybe nit but IMO it was nice when methods matched api paths, at least if they don't do any other particular stuff in the middle.

File: packages/pangea-sdk/pangea/asyncio/services/vault.py [hunk 4]
```diff
@@ -321,719 +329,1422 @@ async def update(
...
+            disabled_at=disabled_at,
             rotation_frequency=rotation_frequency,
             rotation_state=rotation_state,
-            expiration=expiration,
+            rotation_grace_period=rotation_grace_period,
         )
-        return await self.request.post("v1/secret/store", SecretStoreResult, data=input.model_dump(exclude_none=True))
 
-    async def pangea_token_store(
+    async def store_pangea_client_secret(
```
Comment (pangea-andrest) [inline packages/pangea-sdk/pangea/asyncio/services/vault.py:465]: Wait, why you overloaded key_store() and not this one?

File: packages/pangea-sdk/pangea/services/vault/models/asymmetric.py [hunk 5]
```diff
@@ -64,6 +62,108 @@ class VerifyRequest(APIRequestModel):
...
+
+class AsymmetricKeyJwtAlgorithm(str, Enum):
+    """The algorithm of the key for purpose=`jwt`."""
+
+    ES256 = "ES256"
+    ES384 = "ES384"
+    ES512 = "ES512"
+
+
+class AsymmetricKeyPkiAlgorithm(str, Enum):
```
[the remark this answers is not in the document]
Comment (kenany) [reply packages/pangea-sdk/pangea/services/vault/models/asymmetric.py:138]: Is there even enough duplication to be concerned about? Here I think it's just: `ED25519` and the `RSA_PSS_*` ones?

File: packages/pangea-sdk/pangea/asyncio/services/vault.py (hunk 4 above)
Comment (kenany) [reply packages/pangea-sdk/pangea/asyncio/services/vault.py:465]: This one is harder to cleanly overload because Pangea client secret takes 3 input args but secret and Pangea token take only one. And even then, we would need to ask the user to specify an `item_type` in order to differentiate between the types, and at that point it's almost no different from having different named functions. At least with `store_key()` the overloads are combinations across 3+ args (`key_type`, `purpose`, `algorithm`) and help to ensure that e.g. only asymmetric algorithms are specified for asymmetric keys.

File: packages/pangea-sdk/pangea/services/vault/models/keys.py [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    folder: Optional[str] = None
+    metadata: Optional[Metadata] = None
+    tags: Optional[Tags] = None
+    rotation_frequency: Optional[str] = None
+    rotation_state: Optional[RequestRotationState] = RequestRotationState.INHERITED
+    disabled_at: Optional[PangeaDateTime] = None
+    exportable: bool = False
+
+
+class KeysStoreRequest(APIRequestModel):
```
[the remark this answers is not in the document]
Comment (kenany) [reply packages/pangea-sdk/pangea/services/vault/models/keys.py]: Sure, done.

File: packages/pangea-sdk/pangea/asyncio/services/vault.py (hunk 2 above)
Comment (kenany) [reply packages/pangea-sdk/pangea/asyncio/services/vault.py]: Yup, that wasn't available at the time. Done.

File: packages/pangea-sdk/pangea/asyncio/services/vault.py (hunk 3 above)
Comment (kenany) [reply packages/pangea-sdk/pangea/asyncio/services/vault.py:373]: That symmetry can be nice but who names programming functions like that?

File: packages/pangea-sdk/pangea/services/vault/models/asymmetric.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (pangea-andrest) [reply packages/pangea-sdk/pangea/services/vault/models/asymmetric.py:138]: You're right, probably nothing to worry about.

**Review by pangea-andrest — DISMISSED** [review]

File: examples/vault/vault_examples/export.py [hunk 7]
```diff
@@ -23,15 +26,19 @@ def main() -> None:
...
+            name=name,
+            exportable=True,
         )
 
         assert create_response.result
         key_id = create_response.result.id
 
         # export with no encryption
-        exp_resp = vault.export(id=key_id, version=1)
+        exp_resp = vault.export(item_id=key_id, version=1)
```
Comment (pangea-andrest) [inline examples/vault/vault_examples/export.py:41]: why this change? API field is called just `id`

File: examples/vault/vault_examples/get.py [hunk 8]
```diff
@@ -26,10 +27,10 @@ def main() -> None:
 
     try:
         # Fetch the Secure Audit Log token.
-        create_response = vault.get(id=token_id)
+        create_response = vault.get(item_id=token_id)
```
Comment (pangea-andrest) [inline examples/vault/vault_examples/get.py:30]: Oh, I see you changed it here too

**Review by pangea-andrest — APPROVED** [review]
Just minor comment I didn't notice before but it's okey for me.

File: examples/vault/vault_examples/export.py (hunk 7 above)
Comment (kenany) [reply examples/vault/vault_examples/export.py:43]: Most because `id` is a bit generic and it was shadowing the builtin `id()` Python function.
